src/backends/tensorrt/plan_backend.cc

```diff
@@ -1062,6 +1062,12 @@ PlanBackend::Context::Run(
               name_ + "'");
     }
 
+    // Check if all the requested output names in the request header are valid
+    for (const auto& output :
+         payload.request_provider_->RequestHeader().output()) {
+      RETURN_IF_ERROR(base->ValidateOutputName(output.name()));
```

Comment:
You can move this check to `NormalizeRequestHeader()` as this can be detected earlier and ease the load on scheduler and model instances.